### PR TITLE
Revamping workloads page and fixing filter toggle issues

### PR DESCRIPTION
**Pull request by rakol-ms**
Signed-off-by: Rajendra Kolli <rakol@microsoft.com>

workloads_v2

workloads_v2 _filter

### CODE REVIEW

**Comment by rakol-ms** [comment]
Please use the following screenshots as the updated reference

workloads_v2

workloads_v2 _filter

File: src/WebUI/Common/BaseKubeTable.scss [hunk 1]
```diff
@@ -28,6 +20,10 @@
         .kind-tag {
             margin-right: 8px;
         }
+        
+        .bolt-card-header {
```
Comment (madhuv-msft) [inline src/WebUI/Common/BaseKubeTable.scss]: we should see a better rather override, even if we do, we should have a seperate file, so later we can fix all at once.

File: src/WebUI/Common/KubeSummary.tsx [hunk 2]
```diff
@@ -218,6 +213,19 @@ export class KubeSummary extends BaseComponent<IKubeSummaryProps, IKubernetesCon
         this._selectedItemViewMap[SelectedItemKeys.ReplicaSetKey] = (item) => this._getWorkoadPodsViewComponent(item.metadata, item.spec && item.spec.template, item.kind || "ReplicaSet");
     }
 
+    private _getFiterHeaderBar(): JSX.Element {
+        return (
+            <div>
+                <ConditionalChildren renderChildren={this.state.selectedPivotKey === workloadsPivotItemKey || this.state.selectedPivotKey == undefined}>
```
Comment (madhuv-msft) [inline src/WebUI/Common/KubeSummary.tsx]: replace 
this.state.selectedPivotKey == undefined

with
!this.state.selectedPivotKey

File: src/WebUI/Utils.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@
     Licensed under the MIT license.
 */
 
-import { V1ObjectMeta, V1PodStatus, V1PodTemplateSpec } from "@kubernetes/client-node";
+import { V1ObjectMeta, V1PodStatus, V1PodTemplateSpec, V1Container } from "@kubernetes/client-node";
```
Comment (madhuv-msft) [inline src/WebUI/Utils.ts]: remove if you do not need V1Container

File: src/WebUI/WorkLoads/DeploymentsTable.scss [hunk 4]
```diff
@@ -7,3 +7,7 @@
         margin-top: 5px;
     }
 }
+
+.heading-labels {
```
Comment (madhuv-msft) [inline src/WebUI/WorkLoads/DeploymentsTable.scss]: why you want top level css here? if you need keep it in the common file

File: src/WebUI/WorkLoads/DeploymentsTable.tsx [hunk 5]
```diff
@@ -258,14 +259,16 @@ export class DeploymentsTable extends BaseComponent<IDeploymentsTableProperties,
...
+                <span className="kube-flex-row">
+                    <h3>{deployment.metadata.name}</h3>
+                    <div className="heading-labels">
+                        <LabelGroup labelProps={Utils.getUILabelModelArray(deployment.metadata.labels)}
+                            wrappingBehavior={WrappingBehavior.OneLine}
+                            fadeOutOverflow={true}>
+                        </LabelGroup>
+                    </div>
+                </span>
+                <span className="secondary-text kind-tag"> {Resources.DeploymentText} </span>
```
Comment (madhuv-msft) [inline src/WebUI/WorkLoads/DeploymentsTable.tsx]: remove spaces in the text
<span className="secondary-text kind-tag"> {Resources.DeploymentText} </span>

File: src/WebUI/WorkLoads/OtherWorkloadsTable.tsx [hunk 6]
```diff
@@ -0,0 +1,256 @@
+import { V1StatefulSet, V1DaemonSet, V1ReplicaSet } from "@kubernetes/client-node";
+import { BaseComponent, css, format } from "@uifabric/utilities";
```
Comment (madhuv-msft) [inline src/WebUI/WorkLoads/OtherWorkloadsTable.tsx]: format --> use from devops, also do not use for resources text

File: src/WebUI/WorkLoads/OtherWorkloadsTable.tsx [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+    }
+
+    public render(): React.ReactNode {
+        const filteredSet: ISetWorkloadTypeItem[] = this._generateRenderData().filter((set) => {
+            return Utils.filterByName(set.name, this.props.nameFilter);
+        });
+        if (filteredSet.length > 0) {
+            const heading: JSX.Element = (
+                <div className="kube-flex-row" >
+                    <h3>{Resources.OtherWorkloadsText}</h3>
```
Comment (madhuv-msft) [inline src/WebUI/WorkLoads/OtherWorkloadsTable.tsx]: as per figma, we may not want to use h2/h1/h3.

**Comment by rakol-ms** [comment]
WIP images for the k8s screen

workloads_v2

workloads_v2 _filter

**Comment by rakol-ms** [comment]
@madhuv-msft @savenkat @dabasnidhi  PFA the final screenshots of the changes.

File: src/WebUI/Common/BaseKubeTable.tsx [hunk 8]
```diff
@@ -2,19 +2,31 @@
     Copyright (c) Microsoft Corporation. All rights reserved.
     Licensed under the MIT license.
 */
-
-import { TooltipHost, TooltipOverflowMode } from "azure-devops-ui/Tooltip";
-import { BaseComponent, css, IRenderFunction } from "office-ui-fabric-react/lib/Utilities";
+import { BaseComponent, css } from "office-ui-fabric-react/lib/Utilities";
```
Comment (madhuv-msft) [inline src/WebUI/Common/BaseKubeTable.tsx]: do not use office ui, use azuedevops

File: src/WebUI/Common/BaseKubeTable.tsx [hunk 9]
```diff
@@ -24,10 +36,33 @@ export interface ITableComponentProperties<T> extends IVssComponentProperties {
...
-                {this.props.items && this.props.items.length > 0 && this._getComponent()}
-            </div>
+            <CustomCard className={css("flex-grow", "bolt-card-no-vertical-padding", "item-top-padding", "kube-list-content", this.props.className || "")}>
+                {
+                    this.props.headingText &&
+                    <CustomHeader>
+                        <HeaderTitleArea>
+                            <HeaderTitleRow className="kube-flex-row">
+                                {
+                                    (typeof this.props.headingText === 'string') ?
```
Comment (madhuv-msft) [inline src/WebUI/Common/BaseKubeTable.tsx]: double quotes please
'string'

File: src/WebUI/Pods/PodsTable.tsx [hunk 10]
```diff
@@ -122,30 +113,22 @@ export class PodsTable extends BaseComponent<IPodsTableProperties> {
...
-        //         className=""
-        //         excludeTabStop
-        //         href="#"
-        //     >
-        //         {textToRender}
-        //     </Link>
-        // );
+        const itemToRender: React.ReactNode = (
+            <Link
+                className="fontSizeM text-ellipsis bolt-table-link bolt-table-inline-link"
```
Comment (madhuv-msft) [inline src/WebUI/Pods/PodsTable.tsx]: is there a better way, not to hard code these?
className

**Comment by madhuv-msft** [comment]
signing off again.

File: src/WebUI/Pods/PodsTable.tsx (hunk 10 above)
Comment (rakol-ms) [reply src/WebUI/Pods/PodsTable.tsx]: @madhuv-msft I'll take it up in another pass.

**Pull request merged**